src/libcore/num/mod.rs

```diff
@@ -2523,18 +2547,18 @@ macro_rules! same_sign_try_from_int_impl {
     )*}
 }
 
-same_sign_try_from_int_impl!(u128, u8, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i8, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u16, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i16, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u32, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i32, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u64, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i64, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, u128, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, i128, i8, i16, i32, i64, i128, isize);
-same_sign_try_from_int_impl!(u128, usize, u8, u16, u32, u64, u128, usize);
-same_sign_try_from_int_impl!(i128, isize, i8, i16, i32, i64, i128, isize);
+same_sign_try_from_int_impl!(u128, u8, u8, u16, u32, u64, u128, usize, f32, f64);
+same_sign_try_from_int_impl!(i128, i8, i8, i16, i32, i64, i128, isize, f32, f64);
+same_sign_try_from_int_impl!(u128, u16, u8, u16, u32, u64, u128, usize, f32, f64);
+same_sign_try_from_int_impl!(i128, i16, i8, i16, i32, i64, i128, isize, f32, f64);
+same_sign_try_from_int_impl!(u128, u32, u8, u16, u32, u64, u128, usize, f32, f64);
+same_sign_try_from_int_impl!(i128, i32, i8, i16, i32, i64, i128, isize, f32, f64);
+same_sign_try_from_int_impl!(u128, u64, u8, u16, u32, u64, u128, usize, f32, f64);
+same_sign_try_from_int_impl!(i128, i64, i8, i16, i32, i64, i128, isize, f32, f64);
+same_sign_try_from_int_impl!(u128, u128, u8, u16, u32, u64, u128, usize, f32, f64);
```

Comment:
I believe these will trigger #10184 ("float as int" can be UB — consider `u8::try_from(1.0e+100_f64)` which will involve a `(1.0e+100_f64 as u128) > 255` check in the middle).